data/blog/tclip.mdx

```diff
@@ -0,0 +1,137 @@
+<HeroImage
+  width={1280}
+  height={512}
+  desc="An albino pidgeon behind a normal one as found on Oberbaumbrücke in Berlin. Photo by Xe Iaso, 2023."
+  src="/images/tclip/hero.jpg"
+/>
+
+## Installation
+
+The easiest way to use tclip is and to install it on fly.io. You can do that by doing the following:
+
+First, get an authkey from [the admin panel](https://login.tailscale.com/admin/settings/keys). It is a good idea to associate this with the tag `tag:service` or its own `tag:tclip`. Please also do not set the `ephemeral` tag as that will destroy the node when the service shuts down.
+
+Make sure you have MagicDNS enabled [in the admin panel](https://login.tailscale.com/admin/dns), see [here](https://tailscale.com/kb/1081/magicdns/) for more information. This is required for most of the defaults for the `tclip` command and the Emacs package to work.
+
+You will need to have a [fly.io account](https://fly.io/) (this application is designed to fit within their free tier).
+
+### fly.io
+
+In one of your infrastructure management GitHub repositories, create a
+folder for tclip and then copy the following
+[`fly.toml`](https://fly.io/docs/reference/configuration/) template
+into that folder.
+
+```toml
+app = "FLY_APP_NAME"
+
+[build]
+image = "ghcr.io/tailscale-dev/tclip:latest"
+
+[deploy]
+strategy = "immediate"
+
+[env]
+DATA_DIR = "/data"
+
+[mounts]
+source = "tclip_data"
+destination = "/data"
+```
+
+Replace `FLY_APP_NAME` with a name such as `yourorg-tclip` and
+then run these commands with [the `flyctl` command](https://fly.io/docs/hands-on/install-flyctl/):
+
+```
+$ flyctl apps create FLY_APP_NAME
+$ flyctl volumes create tclip_data
+$ flyctl secrets set TS_AUTHKEY=<key>
+$ flyctl deploy
+```
+
+You should be able to open the app at [http://paste](http://paste) and
+paste to your heart's content.
+
+There are more detailed instructions on the [tclip GitHub repo](https://github.com/tailscale-dev/tclip#deploying) if you want to deploy to your own host running Docker or a Linux machine running a modern version of systemd.
+
+## Usage
+
+If you know how to use a web form, you know how to use tclip. Paste your text into the box. Give it a filename. Hit submit. Share the link. There is no next step. It will be available on your tailnet for whoever wants to access it.
+
+If you want to limit access to your tclip server, you can do so with [Tailscale ACL rules](https://tailscale.com/kb/1018/acls/).
+
+All pastes are automatically attributed to the person that posted them, thanks to Tailscale already knowing who you are.
+
+## The Tailscale magic: Funnel integration
+
+However, it doesn't stop there. The real magic comes when we integrate this with [Funnel](https://tailscale.dev/blog/embedded-funnel) so that you can share your pastes publicly. If you enable Funnel egress with the `USE_FUNNEL=true` environment variable (or the `--use-funnel` flag), tclip will register itself publicly with funnel to allow people to view only the following:
```

Comment:
Use a relative link (/blog/embedded-funnel), because its on the site. I think that would work for tailscale.dev, and make the source more readable.